### PR TITLE
Fix issues uncovered by static analysis

### PR DESCRIPTION
**Pull request by leonyu**

### CODE REVIEW

File: src/main/java/ly/bit/nsq/NSQProducer.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public void put(String message) throws NSQException {
 			if (response.getStatusLine().getStatusCode() != 200) {
 				throw new NSQException("POST to " + url + " returned HTTP " + response.getStatusLine().getStatusCode());
 			}
-			if (response != null && response.getEntity() != null) {
```
Comment (leonyu) [inline src/main/java/ly/bit/nsq/NSQProducer.java:90]: `response` was dereferenced 2 lines before, so null check is unnecessary.

File: src/main/java/ly/bit/nsq/lookupd/BasicLookupd.java [hunk 2]
```diff
@@ -21,16 +21,16 @@
 		URL url = null;
 		try {
 			url = new URL(urlString);
+			try {
+				InputStream is = url.openStream();
```
Comment (leonyu) [inline src/main/java/ly/bit/nsq/lookupd/BasicLookupd.java]: an exception on `URL` construction would mean `url` is null

**Comment by leonyu** [comment]
@jehiah please review or ping someone else with Java experience.

**Comment by ploxiln** [comment]
IIRC @davemarchevsky has used java at some point (well technically so have I but not since college so let's ignore that if we can)

**Comment by mreiferson** [comment]
cmon @ploxiln you can do it

**Comment by ploxiln** [comment]
that said... I'm pretty sure you can put it all in one try block, and catch the two exception types in two separate catch clauses... I think that's how try is supposed to be elegant for some stuff

**Comment by leonyu** [comment]
I can walk anyone with Java/.NET experience through, since those are just potential de-referencing errors

**Comment by ploxiln** [comment]
LGTM

**Pull request merged**